test/hotspot/jtreg/compiler/vectorapi/VectorCommutativeOperSharingTest.java

```diff
@@ -0,0 +1,575 @@
+
+    private static int LENGTH = 128;
+    private static final Random RD = Utils.getRandomInstance();
+
+    private static int[] ia;
+    private static int[] ib;
+    private static int[] ir1;
+    private static int[] ir2;
+    private static int[] ir3;
+    private static int[] ir4;
+
+    static {
+        ia = new int[LENGTH];
+        ib = new int[LENGTH];
+        ir1 = new int[LENGTH];
+        ir2 = new int[LENGTH];
+        ir3 = new int[LENGTH];
+        ir4 = new int[LENGTH];
+
+        for (int i = 0; i < LENGTH; i++) {
+            ia[i] = RD.nextInt(Integer.MAX_VALUE);
+            ib[i] = RD.nextInt(Integer.MAX_VALUE);
+        }
+    }
+
+    public static void main(String[] args) {
+        TestFramework.runWithFlags("--add-modules=jdk.incubator.vector");
+    }
+
+    @Test
+    @IR(counts = {IRNode.ADD_VI, IRNode.VECTOR_SIZE_ANY, " 2 ",
+                  IRNode.MUL_VI, IRNode.VECTOR_SIZE_ANY, " 2 ",
+                  IRNode.MAX_VI, IRNode.VECTOR_SIZE_ANY, " 2 ",
+                  IRNode.MIN_VI, IRNode.VECTOR_SIZE_ANY, " 2 "},
+        applyIfCPUFeature = {"avx", "true"})
+    public void testVectorIRSharing1(int index) {
+        IntVector vec1 = IntVector.fromArray(I_SPECIES, ia, index);
+        IntVector vec2 = IntVector.fromArray(I_SPECIES, ib, index);
+        vec1.lanewise(VectorOperators.ADD, vec2)
+            .lanewise(VectorOperators.ADD, vec2.lanewise(VectorOperators.ADD, vec1))
+            .intoArray(ir1, index);
+        vec1.lanewise(VectorOperators.MUL, vec2)
+            .lanewise(VectorOperators.MUL, vec2.lanewise(VectorOperators.MUL, vec1))
+            .intoArray(ir2, index);
+        vec1.lanewise(VectorOperators.MAX, vec2)
+            .lanewise(VectorOperators.MAX, vec2.lanewise(VectorOperators.MAX, vec1))
+            .intoArray(ir3, index);
+        vec1.lanewise(VectorOperators.MIN, vec2)
+            .lanewise(VectorOperators.MIN, vec2.lanewise(VectorOperators.MIN, vec1))
+            .intoArray(ir4, index);
+    }
+
+    @Run(test = "testVectorIRSharing1")
+    public void testVectorIRSharingDriver1() {
+        for (int i = 0; i < I_SPECIES.loopBound(LENGTH); i += I_SPECIES.length()) {
+            testVectorIRSharing1(i);
+        }
+        checkVectorIRSharing1();
+    }
+
+    public void checkVectorIRSharing1() {
+        for (int i = 0; i < I_SPECIES.loopBound(LENGTH); i++) {
+            Verify.checkEQ(ir1[i], (ia[i] + ib[i]) + (ib[i] + ia[i]));
+            Verify.checkEQ(ir2[i], (ia[i] * ib[i]) * (ib[i] * ia[i]));
+            Verify.checkEQ(ir3[i], Integer.max(Integer.max(ia[i], ib[i]), Integer.max(ib[i], ia[i])));
+            Verify.checkEQ(ir4[i], Integer.min(Integer.max(ia[i], ib[i]), Integer.min(ib[i], ia[i])));
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.XOR_VI, IRNode.VECTOR_SIZE_ANY, " 0 ",
+                  IRNode.OR_VI, IRNode.VECTOR_SIZE_ANY, " 1 ",
+                  IRNode.AND_VI, IRNode.VECTOR_SIZE_ANY, " 1 "})
```

Comment:
And this is not going to fail on any architecture? What about architectures other than x64 and aarch64? Do they all support the vector instructions?

I'd rather not that this fails on their machines and then they have to fix it and we have to review their fix again ;)

Reply:
Hi @eme64 ,
The attached script shows me that greatest common divisor of targets supporting IR nodes in this test is x86 :-)
  
[get_c2_IR_supported_targets.txt](https://github.com/user-attachments/files/18572758/get_c2_IR_supported_targets.txt)

![image](https://github.com/user-attachments/assets/daedb8aa-23c5-4c67-b7d3-82859155b88d)